src/functional/ffront/foast_to_itir.py

```diff
@@ -320,6 +320,23 @@ def visit_BinOp(self, node: foast.BinOp, **kwargs) -> itir.FunCall:
             )
         )
 
+    def visit_TernaryExpr(self, node: foast.TernaryExpr, **kwargs) -> itir.FunCall:
+        lowered_node_cond = self.visit(node.condition, **kwargs)
+        lowered_true_expr = self.visit(node.true_expr, **kwargs)
+        lowered_false_expr = self.visit(node.false_expr, **kwargs)
+
+        # Needed in case of sparse or shifted fields
+        if isinstance(node.type, ct.FieldType) and is_local_kind(node.type):
+            return im.call_("if_")(lowered_node_cond, lowered_true_expr, lowered_false_expr)
```

Comment:
This doesn't look quite right even if it might work just fine. According to the spec (https://github.com/GridTools/concepts/blob/master/Iterator-View.md) `if_` only accepts values for its second and third argument, but you pass iterators. Please add a todo for me.

Reply:
@tehrengruber, you're right, it might be something wrong here. Looking at the generated IR of the test with the neighbor sum, it's easy to see that everything is fine in that case because the ternary is applied **inside** the function passed to `reduce`, which will only receive values.

To me, the lowering part is actually correct, but I'm not sure about the condition in the `if`. I think the condition should check if the ternary appears inside a reduction (which effectively ensures that the field is a partially shifted field) to use the reduced lowering path (without the `to_value()` call), and if it's not inside the reduction then use the normal lowering, which will fail for partially shifted fields and therefore likely means that the user code is incorrect...

Does this make any sense?

Reply:
What's the status here?

Reply:
I finally found and fixed the bug here. The problem indeed was in the lowering of the ternary expression inside a reduction. 

The weird `if` case here was only needed because the lowering should be done different inside the reduction but, since this was already taken into account in the `InsideReductionLowering` visitor, why was the weird `if` needed? It turns out there was a typo in the name of the `visit_TernaryExpr` method of `InsideReductionLowering` which meant that it was never called and thus the different lowering path inside reduction never used.

I fixed the typo, remove the unnecessary `if` and now it's good to go.